### PR TITLE
register skeleton provider

### PR DESCRIPTION
**Pull request by BenTheElder**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**: Adds back support for the "skeleton" provider to the e2e tests, used in conformance testing.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**: This broke all of our local cluster testing https://k8s-testgrid.appspot.com/conformance-providerless#Summary https://github.com/kubernetes/kubernetes/compare/8e31d05df...8b36038b4#diff-811d0ade1978c7704546bb4489479a97R46

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by BenTheElder** [comment]
/assign @dims 
working on verifying this locally
cc @Katharine

**Comment by Katharine** [comment]
This fixes it for me.

**Comment by BenTheElder** [comment]
Locally without this patch ginkgo panics while testing skeleton clusters, with it everything is fine. We tell people to use `--provider=skeleton` in a lot of places to side step other baggage we have with `--provider=local` (mostly the bash setup scripts...)

**Review by amwat — COMMENTED** [review]
/lgtm
/hold

**Comment by krzyzacy** [comment]
(you still need to find an approver)

**Review by MrHohn — APPROVED** [review]
/lgtm

**Comment by dims** [comment]
@BenTheElder LGTM

**Comment by spiffxp** [comment]
diff link in the description doesn't work for me, all I know is it was some part of https://github.com/kubernetes/kubernetes/pull/68483

**Pull request merged**
